pkg/api/api.go

```diff
@@ -71,6 +74,7 @@ type Config struct {
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.SkipLabelNameValidationHeader, "api.skip-label-name-validation-header-enabled", false, "Allows to skip label name validation via X-Mimir-SkipLabelNameValidation header on the http write path. Use with caution as it breaks PromQL. Allowing this for external clients allows any client to send invalid label names. After enabling it, requests with a specific HTTP header set to true will not have label names validated.")
 	f.BoolVar(&cfg.EnableOtelMetadataStorage, "distributor.enable-otlp-metadata-storage", true, "If true, store metadata when ingesting metrics via OTLP. This makes metric descriptions and types available for metrics ingested via OTLP.")
+	f.BoolVar(&cfg.GETRequestForIngesterShutdownEnabled, "get-request-for-ingester-shutdown-enabled", false, "Enable GET requests to the /ingester/shutdown endpoint to trigger an ingester shutdown. This is a potentially dangerous operation and should only be enabled consciously.")
```

Comment:
On re-review, this should probably be:
```suggestion
	f.BoolVar(&cfg.GETRequestForIngesterShutdownEnabled, "ingester.get-request-for-shutdown-enabled", false, "Enable GET requests to the /ingester/shutdown endpoint to trigger an ingester shutdown. This is a potentially dangerous operation and should only be enabled consciously.")
```
And the corresponding changes to the docs.

Reply:
Got it, should the `yaml` field name stay the same though?

Reply:
Actually, initially I did as you suggested, but then Peter suggested to make it consistent so I changed it: https://github.com/grafana/mimir/pull/7707#discussion_r1551589188

Reply:
Oh right, I missed that conversation sorry. Looking at the `distributor.enable-otlp-metadata-storage` flag (just above) it exists as both `distributor.enable-otlp-metadata-storage` as the flag and `enable_otel_metadata_translation` as the yaml. @pstibrany Just want to double check your opinion here please.

Reply:
I'm sorry for confusion here. The problem here comes from the fact that config option is defined in `api` config block in YAML, but we're trying to use `ingester.` prefix for CLI flag. At the same time, we're tring to make CLI flag and yaml field name consistent (this was goal of my previous comment, without realizing it's causing more confusion).

I think we should settle on:
* `-api.get-request-for-ingester-shutdown-enabled` for CLI flag
* `get_request_for_ingester_shutdown_enabled` as YAML field name (it's already under `api` section)

If we want to use `-ingester.` prefix for CLI flag, then we should put this into ingester config struct.

Unfortunately I think `enable_otel_metadata_translation` was a mistake, as it doesn't follow our practices. ("enabled" suffix, consistent CLI and YAML names)

Reply:
@jhesketh thanks for catching this. My [suggestion](https://github.com/grafana/mimir/pull/7707#discussion_r1551589188) was not to remove `-ingester.` prefix (only to add `ingester` in the middle of the name), and I missed that this is what happened. 🤦

Reply:
Ah sorry for the misunderstanding too. So should we keep going with the `-ingester.` prefix or `-api.` instead? As you mentioned, I think the latter makes more sense as it's already under the `api` section in the yaml.

Reply:
Hi Kevin, with Josh we agreed to go with `-api.get-request-for-ingester-shutdown-enabled` CLI flag and `get_request_for_ingester_shutdown_enabled` YAML field name. Thank you very much for your patience with us!

Reply:
Hi Peter, no worries! Let me quickly do that